addons/stock/product.py

```diff
@@ -76,7 +76,8 @@ def _get_domain_locations(self, cr, uid, ids, context=None):
             if type(context['location']) == type(1):
                 location_ids = [context['location']]
             elif type(context['location']) in (type(''), type(u'')):
-                domain = [('complete_name','ilike',context['location'])]
+                location = context['location'].replace('/', ' / ')
+                domain = [('complete_name', 'ilike', location)]
```

Comment:
It is not possible to pass just 'Stock' in the name field.  (if you do location.name?)